pkg/controller/controller.go

```diff
@@ -665,18 +675,49 @@ func (c *controller) reconcileInstance(instance *v1alpha1.Instance) {
 			c.recorder.Event(instance, api.EventTypeWarning, errorProvisionCalledReason, s)
 			return
 		}
-		glog.V(5).Infof("Successfully provisioned Instance %v/%v of ServiceClass %v at Broker %v: response: %v", instance.Namespace, instance.Name, serviceClass.Name, broker.Name, response)
 
-		// TODO: process response
+		if respCode == http.StatusAccepted {
+			glog.V(5).Infof("Received asynchronous provisioning response for Instance %v/%v of ServiceClass %v at Broker %v: response: %v", instance.Namespace, instance.Name, serviceClass.Name, broker.Name, response)
+			if response.Operation != "" {
+				instance.Spec.OSBLastOperation = &response.Operation
+			}
 
-		c.updateInstanceCondition(
-			instance,
-			v1alpha1.InstanceConditionReady,
-			v1alpha1.ConditionTrue,
-			successProvisionReason,
-			successProvisionMessage,
-		)
-		c.recorder.Eventf(instance, api.EventTypeNormal, successProvisionReason, successProvisionMessage)
+			// Tag this instance as having an ongoing async operation so we can enforce
+			// no other operations against it can start.
+			c.setAsyncOperationOngoing(instance, true)
+
+			c.updateInstanceCondition(
+				instance,
+				v1alpha1.InstanceConditionReady,
+				v1alpha1.ConditionFalse,
+				asyncProvisioningReason,
+				asyncProvisioningMessage,
+			)
+			c.recorder.Eventf(instance, api.EventTypeNormal, asyncProvisioningReason, asyncProvisioningMessage)
+
+			// Start polling this Service Instance
+			lastOperationRequest := &brokerapi.LastOperationRequest{
+				ServiceID: serviceClass.OSBGUID,
+				PlanID:    servicePlan.OSBGUID,
+			}
+			if response.Operation != "" {
+				lastOperationRequest.Operation = *instance.Spec.OSBLastOperation
+			}
+
+			err = c.poller.CreatePoller(brokerClient, instance, lastOperationRequest, c.handleLastOperationResponse)
```

Comment:
Chainsaw level: I would rather re-queue the instance to be re-reconciled and have logic that polled the endpoint here.  If we run a goroutine per async provision we could potentially exhaust our goroutines.  So, it's better to use a requeue mechanism than to poll in a goroutine.  @sjenning's PR #648 adds the use of a work queue that we could use for instances (and later bindings if/when we support async binding).